src/main.rs

```diff
@@ -164,33 +166,46 @@ async fn main() -> Result<()> {
                 inputs: Input::IpldData(ipld_args),
             };
 
-            let closure_ipld: Ipld = closure.into();
+            let closure_ipld: Ipld = closure.clone().into();
 
             let mut closure_bytes = Vec::new();
             closure_ipld
                 .encode(DagCborCodec, &mut closure_bytes)
                 .expect("CBOR Serialization");
 
-            let closure_cid = Cid::try_from(closure_bytes)?
+            let link: Link<Closure> = Closure::try_into(closure)?;
+            let closure_cid = link
                 .to_string_of_base(Base::Base32HexLower)
                 .expect("string CID");
 
             let mut conn = db::establish_connection();
 
-            let new_receipt = NewReceipt {
+            let new_receipt = Receipt {
+                id: Uuid::new_v4().to_string(),
```

Comment:
Why not let the DB just give you a sequential ID?

Reply:
I guess our other option down the road is to use the CID once there's a signature and nonce, so yeah, okay sure this works for now 👍

Reply:
Yep, I plan on the latter actually, but also to support sequential ID, we won't have varchar. But yeah, plan to use the cid once there's a sig + nonce.